src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -35,20 +38,26 @@ public string HubName
         {
             get
             {
-                if (this.hubName == null)
+                if (this.resolvedHubName == null)
                 {
                     // "WEBSITE_SITE_NAME" is an environment variable used in Azure functions infrastructure. When running locally, this can be
                     // specified in local.settings.json file to avoid being defaulted to "TestHubName"
-                    this.hubName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME") ?? "TestHubName";
-                    this.defaultHubName = this.hubName;
+                    this.resolvedHubName = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME") ?? "TestHubName";
+                    this.defaultHubName = this.resolvedHubName;
+                    this.originalHubName = string.Empty;
                 }
 
-                return this.hubName;
+                return this.resolvedHubName;
             }
 
             set
             {
-                this.hubName = value;
+                if (this.originalHubName == null)
```

Comment:
This approach won't quite work for original hub name if I am not mistaken. When diagnosing the initial bug we had with the new default hub name logic, we realized that the way that the Microsoft libraries for populating Options objects from JSON involves calling the `get` method BEFORE the `set` method. That means that original hub name will always be set to the empty string in real world scenarios.

Reply:
OK, let me revisit this then. Is there a recommended way to reproduce that behavior? For example, will I need to manually run this in the functions host?

Reply:
I just created a nupkg and consumed it in a VS Durable Functions project while debugging that project.

Reply:
I just create a nupkg locally, and then consume that in a VS Durable Functions project with a debugger to step through it.